### PR TITLE
Nginx Proxy manager

### PR DESCRIPTION
**Pull request by DarkYamik**
Created Rockon Nginx Proxy manager using jlesage docker iamge

> To ease and accelerate the PR submission, please use the template below to provide all requested information.
> You can find guidelines on which docker image to use in the [Rockstor's documentation](http://rockstor.com/docs/contribute_rockons.html#which-docker-image-should-i-use), 
> as well as examples of proper formatting in other rock-ons ([here](https://github.com/rockstor/rockon-registry/blob/master/teamspeak3.json)
> and [here](https://github.com/rockstor/rockon-registry/blob/master/nextcloud-official.json))

Fixes # .

### General information on project
This pull request proposes to add a new rock-on for the following project:
- name: Nginx Proxy Manager
- website:  
  https://hub.docker.com/r/jlesage/nginx-proxy-manager/
  https://hub.docker.com/r/jc21/nginx-proxy-manager

- description: Nginx Proxy Manager enables you to easily forward to your websites running at home or otherwise, including free SSL, without having to know too much about Nginx or Letsencrypt.

### Information on docker image
- docker image: <link to image page on docker hub> https://hub.docker.com/r/jc21/nginx-proxy-manager
- is an official docker image available for this project?: Yes, but you need to use docker-compose up -d

### Checklist
- [x] Passes [JSONlint](https://jsonlint.com) validation
- [x] Entry added to `root.json` in _alphabetical_ order (for new rock-on only)
- [x] `"description"` object lists and links to the docker image used
- [x] `"description"` object provides information on the image's particularities (advantage over another existing rock-on for the same project, for instance)
- [x] `"website"` object links to project's main website

### CODE REVIEW

**Comment by FroggyFlox** [comment]
Hi @DarkYamik , 

Thanks a lot for tackling this one, it'll be a very good addition and I'm eager to see it merged. I'll try to get to the review as soon as I can.

Cheers,

**Comment by phillxnet** [comment]
@DarkYamik could you rebase your version of this repository on current master as GitHub is showing a conflict with your root.json addition. This makes it easier to not slip up on my end, as I don't have to manually resolve the conflict, and enables the use of a semi automated system that merges these pull requests here and tests them prior to their merge in the repo itself, and it's consequent publication.

In other words you have probably applied your root.json patch to an older version of this repository and rebasing your version of this repository on current master at GitHub should help all-around. GitHub should then show no conflicts on this page.

Thanks.

File: NginxProxyManager.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+					},
+					"PGID": {
+						"description": "Enter a valid GID to use along with the same UID. It(or the above UID) must have full permissions to all Shares mapped in the previous step.",
+						"label": "GID",
+						"index": 2
+					}
+				}
+			}
+		},
+		"description": "Rockon based on jlesage Docker Image of jc21/nginx-proxy-manager application",
```
Comment (FroggyFlox) [inline NginxProxyManager.json]: I would simply further develop the description and use the "more_info" field to add the first login credentials:
```suggestion
		"description": "Managing Nginx proxy hosts with a simple, powerful interface. <p>Based on jlesage's docker image: <a href='https://hub.docker.com/r/jlesage/nginx-proxy-manager/' target='_blank'>https://hub.docker.com/r/jlesage/nginx-proxy-manager</a>.</p>",
		"more_info": "<h4>First time login</h4><p>Default username is<code>admin@example.com</code>and password is<code>changeme</code>. Make sure to change both the username and password after the first login.</p>",
```

File: NginxProxyManager.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+				"volumes": {
+					"/config": {
+						"description": "Choose a Share for configuration. Eg: create a Share called nginx-proxy-manager-config for this purpose alone.",
+						"label": "Config Storage",
+						"min_size": 1073741824
+					}
+				},
+				"environment": {
+					"PUID": {
+						"description": "UID to run Jdownloader ",
```
Comment (FroggyFlox) [inline NginxProxyManager.json]:
```suggestion
						"description": "UID to run NGinxProxyManager as.",
```

File: NginxProxyManager.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+						"label": "UID",
+						"index": 1
+					},
+					"PGID": {
+						"description": "Enter a valid GID to use along with the same UID. It(or the above UID) must have full permissions to all Shares mapped in the previous step.",
```
Comment (FroggyFlox) [inline NginxProxyManager.json]:
```suggestion
						"description": "Enter a valid GID to use along with the same UID. It (or the above UID) must have full permissions to all shares mapped in the previous step.",
```

File: NginxProxyManager.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+				}
+			}
+		},
+		"description": "Rockon based on jlesage Docker Image of jc21/nginx-proxy-manager application",
+		"ui": {
+			"https": false,
+			"slug": ""
+		},
+		"volume_add_support": true,
+		"website": "https://hub.docker.com/r/jlesage/nginx-proxy-manager/",
```
Comment (FroggyFlox) [inline NginxProxyManager.json]: Link to project's website rather than docker image:
```suggestion
		"website": "https://nginxproxymanager.jc21.com/",
```

**Comment by DarkYamik** [comment]
Hello!

Thank you all for your time, I've already rebased my fork to the last version and now it seems like it has no conflicts.
I applied too all of @FroggyFlox suggestions to the file :)

Thank you all for your time :)

File: NginxProxyManager.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+				"volumes": {
+					"/config": {
+						"description": "Choose a Share for configuration. Eg: create a Share called nginx-proxy-manager-config for this purpose alone.",
+						"label": "Config Storage",
+						"min_size": 1073741824
+					}
+				},
+				"environment": {
+					"PUID": {
+						"description": "UID to run NGinxProxyManager as.",
```
Comment (FroggyFlox) [inline NginxProxyManager.json]: Fix typo:
```suggestion
						"description": "UID to run NginxProxyManager as.",

```

**Comment by DarkYamik** [comment]
Thank you @FroggyFlox !

The typo fix has been committed already :)

If there is anything else that might be modified I will commit it as soon as I can

Thank you!

**Comment by phillxnet** [comment]
@FroggyFlox Thanks for reviewing this Rock-on, much appreciated.

I've had a little trouble in merging it that I'll detail in a bit but merging for now via GitHub and will then test from there as would like to preserve your contributions. From the looks of it a squash would have been nice, I'll explain later.

@DarkYamik This is nice and thank for persevering. No need for a review from me as @FroggyFlox has already done one. I'm going to merge via GitHub for now but there were some problems, mainly down, I think, to the mid pr rebase. The patch file my semi automated test systems pull down from GitHub contain the full history so the pull request really needed to be squashed and re-submitted but that looses @FroggyFlox contribution and messing you around some more so let skip that and get this one in. Also note that it's highly advisable to have a pull request specific branch as that makes it easier to keep your pull requests divided from one another. But lets get this one in as is and take it from there.

Thanks again from me for submitting this pull request, much appreciated. I'll merge now and do some testing from there then publish soon there after if all looks well.

**Comment by phillxnet** [comment]
@DarkYamik and @FroggyFlox Thanks to both of you for getting this done. It looks like an really interesting project that should fit in well with our other Rock-ons.

I, like @FroggyFlox haven't tested it's actual function but it looks to be as intended and we can always tweak what's needed from here on.

Listing:
nginx-proxy-manager-listing
Installed:
nginx-proxy-manager-installed
dashboard:
nginx-proxy-manager-dashboard

Now to my earlier predicament:

My semi automated buildbot tester gave me the following error when I began to test this pr:
```
./211.patch:18: trailing whitespace.
{
./211.patch:19: trailing whitespace.
	"Nginx-proxy-manager": {
./211.patch:20: trailing whitespace.
		"containers": {
./211.patch:21: trailing whitespace.
			"nginx-proxy": {
./211.patch:22: trailing whitespace.
				"image": "jlesage/nginx-proxy-manager",
error: patch failed: root.json:35
error: root.json: patch does not apply
```
And this system uses a GitHub generated patch gained thus:
curl -L https://github.com/rockstor/rockon-registry/pull/211.patch -o 211.patch

and given this patch has the entire history of each commit as it happened, in order, my prior advise on re-basing wasn't enough to have the cleanly apply. But it worked for the end result, hence me merging and then testing. But just to be extra cautions I prefer to be able to merge locally, via the indicated patch and I then auto reconfigure test clients to read this patched version from a local apache. This way I can test end to end prior to actually merging on GitHub.

Anyway just though you might both be interested in the hick-up and if you look at the resulting patch here:
https://github.com/rockstor/rockon-registry/pull/211.patch
you can see how it includes the initial failed patch (pre rebase). Hence a squash, post rebase, would have lost this history and got us a clean patch. But we would also have lost the collaborative nature of this pr. Oh well.

So thanks again @DarkYamik and @FroggyFlox, this looks like a really nice addition. But I have no idea as of yet how to use it :). Maybe a doc pull request to our "Rock-ons with write-ups" section (in a branch of it's own) with a basic setup for one other Rock-on, or the like, would do the trick :).

Happy to have merged and it's now published.

**Comment by mikey0000** [comment]
Sorry to write this here, but has anyone tried routing to the host machine? I cannot get this image to route to the host, e.g another docker image on the same machine.

**Comment by FroggyFlox** [comment]
Hi @mikey0000 ,

I myself have not had the time to set it up for myself yet, but it seems indeed entirely possible to do what you are looking for. I would try on our forum as I believe we do have several users there who are or have been using Nginx Proxy Manager so you should be able to get some help there.

**Comment by DarkYamik** [comment]
Hello @mikey0000 I currently have the setup you indicate on my machine and so far is working correctly. I have a few docker containers running on the same machine as the Nginx one and the traffic gets routed correctly.

Nonetheless, as @FroggyFlox indicated, you might get more help on the forums :)

**Comment by mikey0000** [comment]
I managed to sort it out. Mostly my lack of knowledge on docker networking and reconfiguring nextcloud a little bit.

13/11/2020 12:16:58 am DarkYamik <notifications@github.com>:
